src/app/task-details/task-details.component.ts

```diff
@@ -0,0 +1,87 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { TaskService } from '../service/task.service';
+import { Task } from '../models/task.model';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormBuilder, FormGroup, ReactiveFormsModule, Validators } from '@angular/forms';
+import { Subject, takeUntil } from 'rxjs';
+import { response } from 'express';
+import { HttpClientModule } from '@angular/common/http';
+
+@Component({
+  selector: 'tasko-task-details',
+  standalone: true,
+  imports: [CommonModule, ReactiveFormsModule, HttpClientModule],
+  templateUrl: './task-details.component.html',
+  styleUrl: './task-details.component.scss'
+})
+export class TaskDetailsComponent implements OnInit {
+  task?: Task;
```

Comment:
task should not be undefined, if you can't find a task for a specific ID then it should throw an error or maybe redirect